homeassistant/components/octoprint/__init__.py

```diff
@@ -232,6 +247,16 @@ async def _async_update_data(self):
         else:
             self._printer_offline = False
 
+        # when the printer comes back online, check if there are any new tools that need to be configured
```

Comment:
Note that the coordinator will only do automatic refreshes as long as there are listeners (enabled entities) registered in the coordinator. Maybe we should register a listener that adds the entities instead of using the dispatcher? Otherwise we won't search for new tool entities if the user disables all other entities.

Here's an example from Netgear:
https://github.com/home-assistant/core/blob/5b28e2d9832c4cd90eb9f1204f4f5b1002bf7450/homeassistant/components/netgear/sensor.py#L251

Reply:
TIL, thanks for the tip. I think I updated to match the pattern in the netgear integration